src/uiSelectDirective.js

```diff
@@ -83,6 +83,10 @@ uis.directive('uiSelect',
             $select.sortable = sortable !== undefined ? sortable : uiSelectConfig.sortable;
         });
 
+        scope.$watch(function () { return scope.$eval(attrs.backspaceReset); }, function(newVal) {
```

Comment:
Can you use attrs.$observe(...) here instead? Only $eval when value is different. (See resetSearchInput a few lines down)